lib/types/enums.ts

```diff
@@ -24,6 +24,7 @@ export enum SwapDealStatus {
 }
 
 export enum SwapDealState {
-  Normal = 0,
+  Active = 0,
   Error = 1,
+  Completed = 2,
```

Comment:
Is there a reason we'd want a "completed" state *and* a completed status? It seems like it's not telling us anything different from `SwapDealStatus.SwapCompleted` and I don't think we'd want the redundancy.

Reply:
As I wrote, a state is more `external` while status is internal and deals with the lifecycle of the object. I think we should implement a `state` also in other objects (mainly `order`).  It can be that state and status have similar value.

Reply:
Ok, what do you think about renaming `status` to something like `phase` (and `SwapDealPhase`). That I think would more clearly differentiate it from `state`.

Reply:
No objection.

Reply:
OK cool let's go with that. I'm just looking into that possible race condition you mentioned, then should be good to go.